sympy/matrices/expressions/matpow.py

```diff
@@ -67,7 +67,10 @@ def doit(self, **kwargs):
         elif isinstance(base, MatrixBase) and exp.is_number:
             if exp is S.One:
                 return base
-            return base**exp
+            return base.evaluate_power(exp)
+        from sympy.core import Symbol
+        if isinstance(base, MatrixBase) and type(exp) == type(Symbol('test_symbol')):
```

Comment:
I think I made a typo when trying it before.
Using `isinstance(exp, Symbol)` works now.I'll update my PR if @asmeurer agrees with the other change.